tests/test_activate.py

```diff
@@ -3099,3 +3099,138 @@ def test_activate_and_deactivate_for_uninitialized_env(conda_cli):
     assert conda_error.value.message.startswith(
         "Run 'conda init' before 'conda deactivate'"
     )
+
+
+# The MSYS2_PATH tests are slightly unusual in two regards: firstly
+# they stat(2) for potential directories which indicate which (of
+# several) possible MSYS2 environments have been installed; secondly,
+# conda will pass a Windows pathname prefix but conda-build will pass
+# a Unix pathname prefix (in particular, an MSYS2 pathname).
+
+
+def assert_MSYS2_PATH(create_dirs=None, exp_paths=None, no_paths=None):
+    with tempdir() as td:
+        mkdir_p(join(td, "conda-meta"))
+        activate_d_dir = mkdir_p(join(td, "etc", "conda", "activate.d"))
+        activate_d_1 = join(activate_d_dir, "see-me.sh")
+        activate_d_2 = join(activate_d_dir, "dont-see-me.bat")
+        touch(join(activate_d_1))
+        touch(join(activate_d_2))
+
+        write_state_file(td)
+
+        library_dir = join(td, "Library")
+        mkdir_p(dirname(library_dir))
+
+        if create_dirs:
+            for dir in create_dirs:
+                mkdir_p(dirname(join(library_dir, dir, "bin")))
+
+        # td is a Windows pathname
+        ce_act = CmdExeActivator()
+        builder = ce_act.build_activate(td)
+        paths = ce_act._replace_prefix_in_path(td, td)
+        new_path = ce_act.pathsep_join(paths)
+
+        export_vars, unset_vars = ce_act.get_export_unset_vars(
+            PATH=new_path,
+            CONDA_PREFIX=td,
+            CONDA_SHLVL=1,
+            CONDA_DEFAULT_ENV=td,
+            CONDA_PROMPT_MODIFIER=get_prompt_modifier(td),
+            # write_state_file
+            ENV_ONE=ENV_ONE,
+            ENV_TWO=ENV_TWO,
+            ENV_THREE=ENV_THREE,
+            ENV_WITH_SAME_VALUE=ENV_WITH_SAME_VALUE,
+        )
+        assert builder["unset_vars"] == unset_vars
+        assert builder["export_vars"] == export_vars
+
+        if exp_paths:
+            for dir in exp_paths:
+                exp_dir = join(library_dir, dir, "bin")
+                assert ce_act.path_conversion(exp_dir) in paths
+
+        if no_paths:
+            for dir in no_paths:
+                exp_dir = join(library_dir, dir, "bin")
+                assert ce_act.path_conversion(exp_dir) not in paths
```

Comment:
@ifitchet is this the only part of the test you actually care about? everything above this appears to just be initialization?

Reply:
@kenodegard The tests want to:

1. create a physical environment because
2. they will create any (optional) `create_dirs` before
3. activating the environment (and the whole point is that the environment has different PATHs because of (2))
4. check for the things we (now) expect `exp_paths`
5. check for the absence of things we shouldn't have `no_paths`

Other than that, the rest of it is cut'n'paste but I don't know how much of that is _required_ to bootstrap a physical environment in the test code.  I'm all for cutting it back if we can.